### ISSUE TITLE
No overload matches this call

### ISSUE THREAD

**Issue by 10centmedialabs** [wix/react-native-ui-lib#670]
Having issues again with typescript project. See below, not sure if this is something related to the library (like a missing prop or something), but it's causing my editor to scream :scream:. I see errors everywhere :weary:. Note: Many other components has these errors too. Please advise. I'm fond for code confidence and type safety, which make me use typescript. I encourage you to give typescript a better attention too. Thanks!

<img width="533" alt="Screen Shot 2020-02-21 at 12 25 54 PM" src="https://user-images.githubusercontent.com/59824689/74998959-faa9f200-54a5-11ea-939d-2c50ccd4934f.png">
<img width="564" alt="Screen Shot 2020-02-21 at 12 26 22 PM" src="https://user-images.githubusercontent.com/59824689/74998968-fed60f80-54a5-11ea-836f-dabf454b80af.png">
<img width="609" alt="Screen Shot 2020-02-21 at 12 29 56 PM" src="https://user-images.githubusercontent.com/59824689/74998969-ff6ea600-54a5-11ea-844f-04d6b189aeb3.png">

**Comment by ethanshar** [comment]
We're aiming to integrate typescript soon in our code so these issues will be solved.

**Comment by 10centmedia** [comment]
Thank you. I'd appreciate that a lot :thumbsup:

**Comment by girishjjain** [comment]
Facing same issue, I am using `RadioButton` component as below:
```
<RadioGroup marginT-30 initialValue={'per_month'}>
  <RadioButton value={'checked'} label={'per month'} borderRadius={15} />
  <RadioButton value={'unchecked'} label={'per year'} borderRadius={15} />
</RadioGroup>
```

and getting following error:
```
No overload matches this call.
  Overload 1 of 2, '(props: Readonly<RadioGroupChild<RadioButtonProps> & Partial<Record<ColorName, boolean>> & Partial<Record<TypographyName, boolean>> & Partial<...> & Partial<...> & Partial<...> & Partial<...>>): RadioButton', gave the following error.
    Type '{ label: string; borderRadius: number; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<RadioButton> & Readonly<RadioGroupChild<RadioButtonProps> & ... 5 more ... & Partial<...>> & Readonly<...>'.
      Property 'label' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<RadioButton> & Readonly<RadioGroupChild<RadioButtonProps> & ... 5 more ... & Partial<...>> & Readonly<...>'.
  Overload 2 of 2, '(props: RadioGroupChild<RadioButtonProps> & Partial<Record<ColorName, boolean>> & Partial<Record<TypographyName, boolean>> & Partial<...> & Partial<...> & Partial<...> & Partial<...>, context?: any): RadioButton', gave the following error.
    Type '{ label: string; borderRadius: number; }' is not assignable to type 'IntrinsicAttributes & IntrinsicClassAttributes<RadioButton> & Readonly<RadioGroupChild<RadioButtonProps> & ... 5 more ... & Partial<...>> & Readonly<...>'.
      Property 'label' does not exist on type 'IntrinsicAttributes & IntrinsicClassAttributes<RadioButton> & Readonly<RadioGroupChild<RadioButtonProps> & ... 5 more ... & Partial<...>> & Readonly<...>'.ts(2769)
```

I don't have much experience writing type definitions but if you can give some guidance for how to fix this maybe I can create a PR with the fix. Thanks in advance!

**Comment by ethanshar** [comment]
I would consider `//@ts-ignore` till we get this fixed

**Comment by girishjjain** [comment]
> I would consider `//@ts-ignore` till we get this fixed

Looks like the latest version (am using `"react-native-ui-lib": "^5.5.4",`) has lot of type errors, almost all UI elements are complaining unlike before when few did. I had to disable the whole file using `// @ts-nocheck` at the top of file.

**Comment by ethanshar** [comment]
@Niryo 
I think the recent changes with how we distribute typings break stuff...

**Comment by Niryo** [comment]
@ethanshar yep, it's all the same problem. I am working on a solution for this, hopefully by the end of the day

**Comment by ethanshar** [comment]
should be fixed in 5.6.0
